libcxx/include/__algorithm/iterator_operations.h

```diff
@@ -216,6 +216,9 @@ struct _IterOps<_ClassicAlgPolicy> {
   }
 };
 
+template <class _AlgPolicy, class _Iter>
+using __policy_iter_diff_t = typename _IterOps<_AlgPolicy>::template __difference_type<_Iter>;
```

Comment:
I would find it clearer to instead do this at the beginning of algorithms that need it:

```c++
using _Difference = typename _IterOps<_AlgPolicy>::template __difference_type<_Iter>;
```

And then to use `_Difference` throughout. `__policy_iter_diff_t` introduces yet another short-name that I'd rather not have to remember. Not a blocking comment since this is somewhat subjective.

Reply:
The problem is that you also have to mention `typename _IterOps<_AlgPolicy>::template __difference_type<_Iter>` in the declaration, which is really not easy to read IMO. FWIW I think we may want to rethink the `iterator_operations.h`, since it's become a quite ugly catch-all of different operations.